### PR TITLE
Feature/hubspot integration

### PR DESCRIPTION
**Pull request by Nicolai-Bak**

### CODE REVIEW

**Review by AntonEsbjergIT-Minds — CHANGES_REQUESTED** [review]
It seems that not all entitites from hubspot are saved in the database. AccountManager and Supplier are saved in the database, but deals, clients, locations and contacts are not saved in the database. Also there are a few minor comments.

File: Pelican.Domain/Shared/Result.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+	public static Result<TValue> Failure<TValue>(Error error) => new(default, false, error);
+
+	public static Result<TValue> Create<TValue>(TValue value, Error error)
+		where TValue : class
+		=> value is not null
+		? Success(value)
+		: Failure<TValue>(error);
+
+	public static Result FirstFailureOrSuccess(params Result[] results)
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Domain/Shared/Result.cs:34]: Its like FirstOrDefault or SingleOrDefault - it tries to return a failure but if the arrays does not contain a failure it returns Succes as default

[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Domain/Shared/Result.cs:34]: Ah makes sense

[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Domain/Shared/Result.cs:34]: DoesResultContainFailure sounds like returning a boolean to me

**Pull request merged**